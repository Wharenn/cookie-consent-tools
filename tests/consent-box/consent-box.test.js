import consentBox from '../../src/consent-box/consent-box';

beforeEach(() => {
  document.body.innerHTML = '<div id="container"></div>';
});

describe('ConsentBox', () => {
  test('with default config is visible in the expected container, without see more link, and in message mode', () => {
    consentBox.initialize();

    expect(consentBox.getOptions().autoDisplay).toBeTruthy();
    expect(consentBox.getOptions().type).toBe('message');
    expect(document.getElementById('container')).toMatchSnapshot();
  });

  test('can have a see more link', () => {
    consentBox.initialize({
      seeMoreLink: {
        href: 'https://foo.bar',
        target: '_blank',
      },
    });

    expect(document.getElementById('container')).toMatchSnapshot();
  });

  test('can have a customized see more link', () => {
    consentBox.initialize({
      messages: {
        seeMoreLabel: 'Show me more!',
      },
      seeMoreLink: {
        href: '#',
        target: '_self',
      },
    });

    expect(consentBox.getOptions().messages).toStrictEqual({
      seeMoreLabel: 'Show me more!',
      message: 'This website uses cookies to provide you the best user experience.',
      okButton: 'Got it',
      approveButton: 'Allow Cookies',
      declineButton: 'Decline',
    });
    expect(document.getElementById('container')).toMatchSnapshot();
  });

  test('can have customized messages', () => {
    consentBox.initialize({
      messages: {
        message: 'There is tasty cookies for you.',
        okButton: 'Ok, close it',
        seeMoreLabel: 'Show me more!',
      },
      seeMoreLink: {
        href: '#',
        target: '_self',
      },
    });

    expect(document.getElementById('container')).toMatchSnapshot();
  });

  test('can be set in choice mode', () => {
    expect(consentBox.getOptions().type).toBe('message');

    consentBox.initialize({
      type: 'choice',
    });

    expect(document.getElementById('container')).toMatchSnapshot();
  });

  test('can have customized choice button messages', () => {
    consentBox.initialize({
      messages: {
        approveButton: 'I approve',
        declineButton: 'I decline',
      },
    });

    expect(document.getElementById('container')).toMatchSnapshot();
  });

  test('can have onShow hook', () => {
    const hookFunction = jest.fn();
    consentBox.initialize({
      onShow: hookFunction,
    });

    expect(hookFunction).toHaveBeenCalled();
  });

  test('can have onHide hook', () => {
    const hookFunction = jest.fn();
    consentBox.initialize({
      onHide: hookFunction,
    });

    consentBox.hide();

    expect(hookFunction).toHaveBeenCalled();
  });

  test('can have onOkButtonClick hook', () => {
    const hookFunction = jest.fn();
    consentBox.initialize({
      onOkButtonClick: hookFunction,
    });

    document.querySelector('#consent-box .consentButton.ok').click();

    expect(hookFunction).toHaveBeenCalled();
  });

  test('can have onApproveButtonClick hook', () => {
    const hookFunction = jest.fn();
    consentBox.initialize({
      type: 'choice',
      onApproveButtonClick: hookFunction,
    });

    document.querySelector('#consent-box .consentButton.approve').click();

    expect(hookFunction).toHaveBeenCalled();
  });

  test('can have onDeclineButtonClick hook', () => {
    const hookFunction = jest.fn();
    consentBox.initialize({
      type: 'choice',
      onDeclineButtonClick: hookFunction,
    });

    document.querySelector('#consent-box .consentButton.decline').click();

    expect(hookFunction).toHaveBeenCalled();
  });
});
