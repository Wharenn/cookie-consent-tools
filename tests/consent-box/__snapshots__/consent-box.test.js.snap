// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsentBox can be set in choice mode 1`] = `
<div
  id="container"
>
  <div
    id="consent-box"
  >
    <div
      class="messageContainer"
    >
      <span>
        There is tasty cookies for you. 
      </span>
      <a
        href="#"
        target="_self"
      >
        Show me more!
      </a>
    </div>
    <div
      class="buttonContainer"
    >
      <a
        class="consentButton decline"
        href="#"
      >
        Decline
      </a>
      <a
        class="consentButton approve"
        href="#"
      >
        Allow Cookies
      </a>
    </div>
  </div>
</div>
`;

exports[`ConsentBox can have a customized see more link 1`] = `
<div
  id="container"
>
  <div
    id="consent-box"
  >
    <div
      class="messageContainer"
    >
      <span>
        This website uses cookies to provide you the best user experience. 
      </span>
      <a
        href="#"
        target="_self"
      >
        Show me more!
      </a>
    </div>
    <div
      class="buttonContainer"
    >
      <a
        class="consentButton ok"
        href="#"
      >
        Got it
      </a>
    </div>
  </div>
</div>
`;

exports[`ConsentBox can have a see more link 1`] = `
<div
  id="container"
>
  <div
    id="consent-box"
  >
    <div
      class="messageContainer"
    >
      <span>
        This website uses cookies to provide you the best user experience. 
      </span>
      <a
        href="https://foo.bar"
        target="_blank"
      >
        See more...
      </a>
    </div>
    <div
      class="buttonContainer"
    >
      <a
        class="consentButton ok"
        href="#"
      >
        Got it
      </a>
    </div>
  </div>
</div>
`;

exports[`ConsentBox can have customized choice button messages 1`] = `
<div
  id="container"
>
  <div
    id="consent-box"
  >
    <div
      class="messageContainer"
    >
      <span>
        There is tasty cookies for you. 
      </span>
      <a
        href="#"
        target="_self"
      >
        Show me more!
      </a>
    </div>
    <div
      class="buttonContainer"
    >
      <a
        class="consentButton ok"
        href="#"
      >
        Ok, close it
      </a>
    </div>
  </div>
</div>
`;

exports[`ConsentBox can have customized messages 1`] = `
<div
  id="container"
>
  <div
    id="consent-box"
  >
    <div
      class="messageContainer"
    >
      <span>
        There is tasty cookies for you. 
      </span>
      <a
        href="#"
        target="_self"
      >
        Show me more!
      </a>
    </div>
    <div
      class="buttonContainer"
    >
      <a
        class="consentButton ok"
        href="#"
      >
        Ok, close it
      </a>
    </div>
  </div>
</div>
`;

exports[`ConsentBox with default config is visible in the expected container, without see more link, and in message mode 1`] = `
<div
  id="container"
>
  <div
    id="consent-box"
  >
    <div
      class="messageContainer"
    >
      <span>
        This website uses cookies to provide you the best user experience. 
      </span>
    </div>
    <div
      class="buttonContainer"
    >
      <a
        class="consentButton ok"
        href="#"
      >
        Got it
      </a>
    </div>
  </div>
</div>
`;
